File: NG-Training/NG.NET-Apps/Knowledge-Base-App/src/app/interfaces/compliance.ts
import { Presentation } from "./presentation";
import { Question } from "./question";

export interface compliance {
    id?: string;
    title: string;
    description: string;
    requiredPercentage: number;
    createdBy: string;
    createdDate: Date;

    presentation: Presentation;

    questions: Question[];
}

File: .NET-Training/API-Training/StandardApiSolution/KnowledgeBase/Models/Compliance.cs
namespace TaskTracker.Models
{
    public class Compliance
    {
        public Guid Id { get; set; } = Guid.NewGuid();
        public string Title { get; set; }
        public string Description { get; set; }
        public double RequiredPercentage { get; set; }
        public string CreatedBy { get; set; }
        public DateTime CreatedDate { get; set; }

        public Presentation Presentation { get; set; }
        
        public List<Question> Questions { get; set; }
    }
}
